### ISSUE TITLE
Auth requires MFA even when MFA is turned off but SMS phone number validation is on

### ISSUE THREAD

**Issue by poppybank** [aws-amplify/amplify-cli#887]
** Which Category is your question related to? **
auth

** What AWS Services are you utilizing? **
Cognito user pool auth

** Provide additional details e.g. code snippets **
amplify --version: 1.1.0

Auth is configured to have email address be optional and phone number is required so validation of the phone number is possible through SMS.  

MFA is configured to be off but since it need to validate the phone number with SMS, it seems to configure the mfaTypes to be SMS Text Message at which point, even if mfaConfiguration is off, it still requires MFA auth every time the app launches.

    "autoVerifiedAttributes": [
        "phone_number"
    ],
    "mfaConfiguration": "OFF",
    "mfaTypes": [
        "SMS Text Message"
    ],
     "requiredAttributes": [
        "phone_number"
    ],

I would to be able to have the phone number validation but no MFA at all. Is there a way to make this happen?

**Comment by attilah** [comment]
It seems strange to me, since I've MFA off and email, phone number is required and I can login my users with email+password, no MFA pops up, just for registration, since phone number verification is required.

I'd double check my cognito pool settings, what was pushed, because if provisioned cognito settings are looking good, then it is not an Amplify CLI issue, but a cognito one.

**Comment by kaustavghosh06** [comment]
@poppybank Could you tell me the steps you took in the CLI when adding auth to your project?

**Comment by poppybank** [comment]
@attilah The difference here is trying to login with phone number and password with no email. This may be a Cognito issue but the thing I'm trying to get to is being able to enter either a username or a phone number in the sign in field (+password) and have SMS verification of the phone number but no MFA. 

For the Amplify CLI issue, I'm wondering what I'd need to set up to enable what I _think_ is the Username option and second and third checkbox in this image.

![image](https://user-images.githubusercontent.com/7173455/53038869-9982b500-3432-11e9-9ca9-e807ff73a562.png)


I can check at aws/amazon-cognito-auth-js about the MFA issue since MGA is listed as off yet the sign in flow does still pop up the MGA screen (with a skip option --- weird) whenever I reload the React-Native app and that is very annoying.

**Comment by poppybank** [comment]
@kaustavghosh06 I first added Analytics which automatically added Cognito.  Then I did an` amplify update auth` option to set up the defaults and finally edited the XXX-cloudformation-template.yml file to add 
`     UsernameAttributes:
        - "phone_number"
`
and 
`Schema: 
        -
          Name: phone_number
          Required: true
          Mutable: true`
and then did `amplify push`

**Comment by poppybank** [comment]
@kaustavghosh06 here's the full UserPool section:

`  UserPool:
  # Created upon user selection
  # Depends on SNS Role for Arn if MFA is enabled
    Type: AWS::Cognito::UserPool
    UpdateReplacePolicy: Retain
    Properties:
      UsernameAttributes:
        - "phone_number"
      UserPoolName: !If [ShouldNotCreateEnvResources, !Ref userPoolName, !Join ['',[!Ref userPoolName, '-', !Ref env]]]
      Schema: 
        
        -
          Name: phone_number
          Required: true
          Mutable: true
        
      
      AutoVerifiedAttributes: !Ref autoVerifiedAttributes
      
            
      Policies:
        PasswordPolicy:
          MinimumLength: !Ref passwordPolicyMinLength
          RequireLowercase: true
          RequireNumbers: true
          RequireSymbols: false
          RequireUppercase: true
      MfaConfiguration: !Ref mfaConfiguration
      SmsVerificationMessage: !Ref smsVerificationMessage
      SmsConfiguration: 
        SnsCallerArn: !GetAtt SNSRole.Arn
        ExternalId: !Ref roleExternalId 
 `

**Comment by sbe-arg** [comment]
I see this same issue only after enabling SMS then trying to disable it via CLI for a user with verified phone, the user SMS MFA remains on and can only be disabled via console.

cognito UI shows MFA disabled and the user is being prompted, have to enable then re disable MFA for SMS prompt to go off.

**Comment by cjihrig** [comment]
Closing as this as been labeled as "won't fix."
